### PR TITLE
Tagging Attribute Storage Policy properly 

### PR DESCRIPTION
**Pull request by paulr34**
cleaning up storage policy code and fixing issue_1110 in the process https://github.com/project-chip/zap/issues/1110

### CODE REVIEW

File: src-electron/db/query-config.js [hunk 1]
```diff
@@ -245,7 +245,7 @@ INTO ENDPOINT_TYPE_ATTRIBUTE (
       cluster.endpointTypeClusterId,
       attributeId,
       staticAttribute.defaultValue ? staticAttribute.defaultValue : '',
-      dbEnum.storageOption.ram,
+      staticAttribute.storagePolicy,
```
Comment (brdandu) [inline src-electron/db/query-config.js:248]: What is staticAttribute?

**Comment by brdandu** [comment]
Should this optionality come from the xml for certain attributes?

File: src-electron/zcl/zcl-loader-silabs.js [hunk 2]
```diff
@@ -556,15 +556,15 @@ function prepareCluster(cluster, context, isExtension = false) {
           attribute.$.reportingPolicy
         )
       }
-      let storagePolicy = dbEnum.storagePolicy.any
+      let storagePolicy = dbEnum.storageOption.ram
       if (context.listsUseAttributeAccessInterface && attribute.$.entryType) {
-        storagePolicy = dbEnum.storagePolicy.attributeAccessInterface
+        storagePolicy = dbEnum.storageOption.external
```
Comment (brdandu) [inline src-electron/zcl/zcl-loader-silabs.js:561]: Is this correct for all use cases @bzbarsky-apple ?

File: src-electron/db/query-config.js (hunk 1 above)
Comment (paulr34) [reply src-electron/db/query-config.js:248]: it's an object returned from a query that retrieves attribute data

File: src-electron/zcl/zcl-loader-silabs.js (hunk 2 above)
Comment (paulr34) [reply src-electron/zcl/zcl-loader-silabs.js:561]: It is just a bug fix so the database does not have "attributeAccessInterface" as the entry (see photo) but rather "external" which is what t is supposed to be if it is in that list.
Screenshot 2023-09-14 at 10 50 11 AM

**Comment by paulr34** [comment]
> Should this optionality come from the xml for certain attributes?

We have a list in zcl.json called  "attributeAccessInterfaceAttributes" which identifies cluster/attribute pairs which should be deemed "external" for storage policy.

**Comment by paulr34** [comment]
@brdandu I have to fix the conditional so I have to make one more change

File: src-electron/zcl/zcl-loader-silabs.js (hunk 2 above)
Comment (bzbarsky-apple) [reply src-electron/zcl/zcl-loader-silabs.js]: I'm a little confused.  storagePolicy and storageOption are not the same thing.  Why are we storing storageOption values in storagePolicy @paulr34 ?

**Pull request merged**